### PR TITLE
[backend/frontend] fromOrTo relations filters with different operators/modes combinations (#6390)

### LINKED ISSUE #6390: Some filters in the entities' "knowledge" tabs don't seem to work.
## Description

When I try to filter entities in the timelines of my knowledge tab, nothing changes.

## Environment

OCTI 6.0.7

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Go to the intrusion set APT29
2. Go to the knowledge tab
3. Scroll down to see the timeline
4. Try to filter the type of entity displayed with the "target type" filter

## Screenshots

Screenshot 2024-03-19 162630

### PR DESCRIPTION
**Pull request by Archidoit**
PR content:
- issue: https://github.com/OpenCTI-Platform/opencti/issues/6390
- fix elementWithTargetTypes filter with not_eq and nil operators
- fix fromOrToId filter with complex operators/modes combinations
- fromTypes / toTypes / fromId / toId with empty / not_empty operator

### CODE REVIEW

**Comment by JeremyCloarec** [comment]
Not equals doesn't seem to work properly with related entity type
image

**Comment by JeremyCloarec** [comment]
Same thing for the "empty" operator, we still get results with it select:
image

**Comment by JeremyCloarec** [comment]
The target entity list seem to contain more entities than the ones contained in the timeline
image

**Comment by Archidoit** [comment]
> The target entity list seem to contain more entities than the ones contained in the timeline image

it's normal, all the entities of the platform are present in the filter list values

**Comment by Archidoit** [comment]
> Not equals doesn't seem to work properly with related entity type image

fixed

**Comment by Archidoit** [comment]
all the comments have been fixed :)

File: opencti-platform/opencti-graphql/src/utils/filtering/filtering-constants.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export const complexConversionFilterKeys = [
   SOURCE_RELIABILITY_FILTER, // reliability of the author
   COMPUTED_RELIABILITY_FILTER, // reliability, or reliabilityof the author if no reliability
   INSTANCE_RELATION_FILTER, // nested relation for the from or to of a relationship
-  INSTANCE_FILTER_TARGET_TYPES, // nested relation for the from or type type of a relationship
+  INSTANCE_RELATION_TYPES_FILTER, // nested relation for the from or type type of a relationship
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/utils/filtering/filtering-constants.ts]: small typo ("type type" => "to type")

**Review by JeremyCloarec — APPROVED** [review]

File: opencti-platform/opencti-graphql/src/utils/filtering/filtering-constants.ts (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/utils/filtering/filtering-constants.ts]: fixed

**Pull request merged**